deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/CheckDeployEnvStatusWorker.java

```diff
@@ -0,0 +1,127 @@
+package com.pinterest.teletraan.worker;
+
+import com.pinterest.deployservice.ServiceContext;
+import com.pinterest.deployservice.bean.EnvironBean;
+import com.pinterest.deployservice.dao.DeployDAO;
+import com.pinterest.deployservice.dao.EnvironDAO;
+import com.pinterest.deployservice.dao.UtilDAO;
+import com.pinterest.deployservice.bean.DeployFilterBean;
+import com.pinterest.deployservice.db.DeployQueryFilter;
+import com.pinterest.deployservice.bean.DeployQueryResultBean;
+import com.pinterest.deployservice.bean.*;
+import org.quartz.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.DataOutputStream;
+import java.net.Socket;
+import java.sql.Connection;
+import java.util.*;
+
+public class CheckDeployEnvStatusWorker implements Job {
+    private static final Logger LOG = LoggerFactory.getLogger(CheckDeployEnvStatusWorker.class);
+    private EnvironDAO environDAO;
+    private DeployDAO deployDAO;
+    private UtilDAO utilDAO;
+    public static final int  METRICS_AGENT_PORT = 18126;
+
+    public CheckDeployEnvStatusWorker() {
+
+    }
+
+    void processDeploys() throws Exception {
+        List<String> envIds = environDAO.getAllEnvIds();
+        Collections.shuffle(envIds);
+        //count for different deploy states, need emit them to statsboard
+        int runningDeploys = 0;
```

Comment:
Counters for each deploy state - can we create a class to encapsulate the counter, similar to the "DeployState"? For example we can have DeployCount.RUNNING, DeployCount.FAILING, ...